lxd/db/cluster/update.go

```diff
@@ -65,6 +65,49 @@ var updates = map[int]schema.Update{
 	26: updateFromV25,
 	27: updateFromV26,
 	28: updateFromV27,
+	29: updateFromV28,
+}
+
+// Add content type field to storage volumes
+func updateFromV28(tx *sql.Tx) error {
+	stmts := `ALTER TABLE storage_volumes ADD COLUMN content_type INTEGER NOT NULL DEFAULT 0;
+UPDATE storage_volumes SET content_type = 1 WHERE type = 3;
```

Comment:
We probably should do this a bit better and also populate the content_type properly for all volumes, not just custom volumes.

 - All containers should be of type filesystem
 - All VMs should be of type block
 - All existing custom volumes should be of type filesystem
 - Images is a bit trickier as we need to lookup their type in the image table to know if block (vm) or filesystem (container).

Reply:
> All containers should be of type filesystem

That's the default value, so we don't have to explicitly change anything.

> All VMs should be of type block

That's what the `UPDATE` statement does.

> All existing custom volumes should be of type filesystem

Same as the first. They will automatically be of type `filesystem`.

> Images is a bit trickier as we need to lookup their type in the image table to know if block (vm) or filesystem (container).

Images aren't part of the `storage_volumes` table. Why would we need to change anything regarding images?

Reply:
Images are very much part of the storage_volumes table.

Reply:
```
stgraber@castiana:~$ lxd sql global "SELECT * FROM storage_volumes WHERE type=1;"
+------+------------------------------------------------------------------+-----------------+---------+------+-------------+------------+
|  id  |                               name                               | storage_pool_id | node_id | type | description | project_id |
+------+------------------------------------------------------------------+-----------------+---------+------+-------------+------------+
| 3767 | 50b31a66e61c74d916c0bc6d9df570aba2242644a15e215cd2040c0f7ec9ee42 | 1               | 1       | 1    |             | 1          |
| 4470 | 1bb3c2f730c56213021affc87e7d25f2c25b5186b72b2ca43cd078c04c222157 | 1               | 1       | 1    |             | 1          |
| 4631 | 0125879e56c09a0a6eacdfc16592c15d690803b3599f62b2a3ec1c8914eeb23c | 1               | 1       | 1    |             | 1          |
| 4641 | 5455d3b9ce6702094c15df33ab53a7e30238e6de143943f9bf061c9cad78f4f0 | 1               | 1       | 1    |             | 1          |
| 4645 | 7ca3f048ab4e8846a5ddd10cdc1e1558a2f1bb427e0a023ca9251e52b89b3f2d | 1               | 1       | 1    |             | 1          |
| 4646 | 48178524ecd3e5a15d7f8ac1af6802961015cb3a2d13252a1d1ef96c0577929d | 1               | 1       | 1    |             | 1          |
+------+------------------------------------------------------------------+-----------------+---------+------+-------------+------------+
stgraber@castiana:~$ 
```

Reply:
And they can be of either type depending on whether it's a VM or container image.

Reply:
Right. When using only the `dir` storage backend, they aren't in the DB. That's the reason I didn't see any images.

Reply:
@monstermunchkin a storage volume record is only created for images when the storage driver supports optimized volumes (snapshots basically).